packages/venia-concept/src/components/Button/index.js

```diff
@@ -0,0 +1,31 @@
+import { Component, createElement } from 'react';
+import PropTypes from 'prop-types';
+
+import classify from 'src/classify';
+import defaultClasses from './button.css';
+
+class Button extends Component {
+    static propTypes = {
+        classes: PropTypes.shape({
+            content: PropTypes.string,
+            root: PropTypes.string
+        }),
+        type: PropTypes.oneOf(['button', 'reset', 'submit'])
+    };
+
+    static defaultProps = {
+        type: 'button'
+    };
+
+    render() {
+        const { children, classes, type, ...restProps } = this.props;
+
+        return (
+            <button className={classes.root} type={type} {...restProps}>
+                <span className={classes.content}>{children}</span>
```

Comment:
Is the wrapper `span` needed? Might make more sense to let the consumer decide all the children

Reply:
Yes, unfortunately. In fact, that `span` is the only reason `Button` is a component.

- `<button>` can't be a flex or grid container in some browsers
- `<button>` has a lot of semantic value, so we still want to use it
- `<button>` can only contain phrasing content, not flow content, so no `<div>`
- so we have to wrap content in `<span>`
-

Reply:
Ah cool, missed that. Thanks for clarifying 👍